### DISCUSSION TITLE
Remix appending comma to end of HTML bodies when loader is present

### DISCUSSION THREAD

**Discussion by cill-i-am** [remix-run/remix#10123] [Q&A]
Bit of a strange one here. So we have had this comma appended to the end of our `<body/>` and I think it is caused by Remix. It occurs on all pages, unless a page does not have a dedicated loader **and** has a clientLoader - for some reason none of the below code occurs then. I am really at a loss as to how to fix this, so any help would be very appreciated 

The injected code looks like this:
```TS
<script>window.__remixContext.streamController.enqueue("AN_ARRAY_WITH_A_LOAD_OF_LOADER_DATA");</script>
<!--$-->
<script>window.__remixContext.streamController.close();</script>
<!--/$-->
<!--/$-->
","
<script>function $RC(a,b){a=document.getElementById(a);b=document.getElementById(b);b.parentNode.removeChild(b);if(a){a=a.previousSibling;var f=a.parentNode,c=a.nextSibling,e=0;do{if(c&&8===c.nodeType){var d=c.data;if("/$"===d)if(0===e)break;else e--;else"$"!==d&&"$?"!==d&&"$!"!==d||e++}d=c.nextSibling;f.removeChild(c);c=d}while(c);for(;b.firstChild;)f.insertBefore(b.firstChild,c);a.data="$";a._reactRetry&&a._reactRetry()}};$RC("B:0","S:0")</script>
<script>$RC("B:1","S:1")</script>
```

And here is an example of it:
![Screenshot 2024-10-16 at 16 32 15](https://github.com/user-attachments/assets/347ab7bd-d047-4f3a-a018-c20a476a43fd)
Even removing the outlet etc from the route results in the same:
![Screenshot_2024-10-15_at_15 57 43](https://github.com/user-attachments/assets/72f654c8-7641-4638-b7b3-f5defdef3332)
![Screenshot_2024-10-15_at_15 58 14](https://github.com/user-attachments/assets/3ec73aa7-034f-41e4-8771-966a51b9a8b1)

**Comment by cill-i-am** [comment]
Here's a pared back example of what I mean. The only loader that is run on the `/test` route is the root loader. Notice how adding a client action removes the comma? Otherwise it remains no matter what I do

https://github.com/user-attachments/assets/0a4bdf3f-4c6c-4ce5-a35d-a23b820de5a2

**Comment by chw741852** [comment]
I had the same problem in my project and I found an extra comma in the `entry.server.tsx`  
![WechatIMG408](https://github.com/user-attachments/assets/cfe974ee-72c6-41d5-a3ef-dbd836b13a50)
